src/Types/strings.ts

```diff
@@ -0,0 +1,29 @@
+// Script Filename
+export type ScriptFilename = string /*& { __type: "ScriptFilename" }*/;
+/*export function isScriptFilename(value: string): value is ScriptFilename {
+  // implementation
+}*/
+/*export function sanitizeScriptFilename(filename: string): ScriptFilename {
+  // implementation
+}*/
+export function scriptFilenameFromImport(importPath: string, ns1?: boolean): ScriptFilename {
+  if (importPath.startsWith("./")) importPath = importPath.substring(2);
```

Comment:
Why is our directory handling so bad. "./foo.js" in a script means "/foo.js" (I.e. it's absolute, despite looking relative) and "/foo.js" isn't even allowed.